### PR TITLE
Support starting console maximized.

### PR DESCRIPTION
**Pull request by ipankajg**
This is useful in cases where the window size is not resizable but you
still want to have main console window maximized. The reason for not using
resizable window in this case is fat borders around resizable windows in
windows 8.1.

This image shows the new setting for initial maximized window.
newsetting

This image shows the difference between resizable window border and non-resizable window border. The resizable window has thick border.
resizablevsnonresizableborder

You can try this feature using binaries in this archive.
[console.tar.gz](https://github.com/cbucher/console/files/136203/console.tar.gz)

### CODE REVIEW

**Comment by ipankajg** [comment]
Please let me know if this change can be merged. I have another change for split bar size with new settings in styles that I would submit after this.

**Comment by cbucher** [comment]
I don't understand why "snap to edge" is disabled.

I don't understand why you disable the initial size of the normal window.

To start maximized, you can simply configure a shortcut to console.exe:
Right-click the shortcut > Properties > Shortcut tab > Run

By the way, with shortcut or code, there is a bug when an initial size is specified.
capture

**Comment by ipankajg** [comment]
The shortcut way to maximize the console does not work if you right click on pinned item in taskbar and launch a different tab from jumplist. This is why I thought giving an option to always launch maximized is good. Let me know if there is another way to launch from jumplist and start with maximized size.

The reason I thought to disable those three options was, once you decide to launch a window as maximized i.e. select initial size maximized, then it doesn't make sense to set initial size or initial position or snap to desktop edge because the window would be maximized to begin with.

Basically what I was thinking is that if the window is resizable, then there is no point of selecting Initial size maximized and a non-resizable window will never be resized after initial launch. Anyways, I can keep those options enabled if you think that would be a better choice (even though on a non-resizable initial size maximized window, those options will have no effect).

**Comment by ipankajg** [comment]
Yes, I see the bug where if initial size is specified then somehow maximized setting don't work and window goes partially off the desktop. That seems to be an existing issue but i would see if I can fix it, if you think this change makes sense.

On the other hand, if there is another way to start tabs from jumplist as maximized (i.e. without -reuse flag which is another bug), then i can drop this PR.

**Comment by cbucher** [comment]
> The reason I thought to disable those three options was, once you decide to launch a window as maximized i.e. select initial size maximized, then it doesn't make sense to set initial size or initial position or snap to desktop edge because the window would be maximized to begin with.

You must clarify your goal:
- simply allow to start maximized, in this case don't forget that the normal size! (like I've done with full screen).
- create a normal size that correspond to the maximized size

The label "initial size maximized" is not clear.

**Comment by cbucher** [comment]
>On the other hand, if there is another way to start tabs from jumplist as maximized (i.e. without -reuse flag which is another bug), then i can drop this PR.

There, you lost me. You don't want reuse ConsoleZ instance but you want start new instances maximized. You will have multiple instance maximized and can see only one at time. Why don't reuse the same ConsoleZ instance and use tabs?

**Comment by ipankajg** [comment]
My goal is: "simply allow to start maximized"
I will look at what is needed to do "in this case don't forget that the normal size! (like I've done with full screen)."
I can also enable various options i.e. not disable them when initial size maximized is specified.

"Why don't reuse the same ConsoleZ instance and use tabs?"
There are multiple reasons to use separate windows over tabs
- It allows me to use these windows on multiple monitors
- I can continue to use my regular workflow of alt-tab to switch windows
- I can quickly switch windows using mouse from the taskbar

Depending upon one's usage, this is quite useful.

**Comment by cbucher** [comment]
>It allows me to use these windows on multiple monitors

So you move the window from a monitor to another. So, interest of maximized window at start disappears.

>I can continue to use my regular workflow of alt-tab to switch windows

You can select a tab in one hotkey (or two if ConsoleZ as no focus).
I don't critic your usage, but I cannot validate this reason.

>I can quickly switch windows using mouse from the taskbar

You can quickly switch tab using mouse from the tabs bar.
I don't critic your usage, but I cannot validate this reason.

**Comment by ipankajg** [comment]
"So, interest of maximized window at start disappears."
It does not, you can move a maximized window using win+shift+arrow keys.

Rest of the comments are subjective so I won't reply to them. If you feel the patch is useful, you can incorporate or if any changes are needed, i can do those. Otherwise I can continue using my fork for my personal usage. It is your call.

**Comment by cbucher** [comment]
OK, what do you think of this:

Initial state

That's consistent with others `Position` options.

**Comment by ipankajg** [comment]
I agree, this is better and provides a more consistent interface.

**Comment by cbucher** [comment]
implemented by myself
you can try the last experimental version

**Comment by ipankajg** [comment]
Thank you for implementing this!
